Reorganized install-guide (#470)
MicrosoftDocs/quantum-docs-pr: articles/install-guide/setup/index.md

---
title: Setup Overview
author: gillenhaalb

> ms.author: gillenhaalb@gmail.com

Author: Makes sense, I'll make that change when it comes.

## Install the Quantum Development Kit for Your Preferred Environment ##

### Visual Studio Editors and the Command Line ###

With the .NET core SDK now installed, you can implement programs in Q# directly in either the command line, or editors such as Visual Studio Code and Visual Studio. To get started, install the Quantum Development Kit Samples and Libraries following these instructions for your preferred environment (links to download the corresponding editors are included within):
* [Installing the QDK for Visual Studio Code](xref:microsoft.quantum.install.vs-code)
* [Installing the QDK for Visual Studio](xref:microsoft.quantum.install.vs-2017)
* [Installing the QDK in the command line](xref:microsoft.quantum.install.cmd-line)

### Classical Hosts ###

> If, however, you prefer to work within a classical host environment, we provide interoperability allowing you to implement Q# quantum algorithms using C#, Python, and Jupyter Notebooks. Details and instructions are provided here:

Author: No apologies necessary--that was one of the nagging confusions I had in the back of my mind, and it was tough to discern from merely reading the other docs. This is hugely helpful, thanks.

---
title: Setup Overview
author: gillenhaalb
ms.author: gillenhaalb@gmail.com
ms.date: 7/19/2019
ms.topic: article
uid: microsoft.quantum.install.setup
---

# Setup Overview #

To begin writing and running quantum programs on your own computer, follow these instructions for installing the Quantum Development Kit.

> Whether on Windows, Linux, or macOS, you will need the right machinery in the background, so the first step below is to download the [.NET Core SDK](https://dotnet.microsoft.com/).

Reviewer: I'm not sure this line is necessary, since the instructions directly follow. Is the intent to state that the QDK is supported across all 3 OS's?

Reviewer 2: At least part of my thinking with suggesting this paragraph was that the reference rendering looked really awkward with a `#` header immediately followed by a `##` header. My hope was that some slight text helps space things up a bit and flow a bit better.

> Then, our instructions will guide you through installing the Quantum Devleopment Kit for whichever classical host environment and language you prefer--the choice is yours, as the power of Q# will be at your fingertips regardless!

Reviewer: If removing the previous line, I would remove this as well.

## Install .NET Core SDK ##

> The Quantum Development Kit uses the [.NET Core SDK](https://dotnet.microsoft.com/) to make it easy to create, build, and run Q# projects from the command line.

Reviewer: ```suggestion
The Quantum Development Kit uses the [.NET Core SDK](https://dotnet.microsoft.com/) to make it easy to create, build, and run Q# projects.
```

Whatever your preferred editor or environment may be, to start working with Q# on your own computer, you will need the latest version (2.2 or later) installed.
You can do this by following the instructions from the [.NET downloads page](https://www.microsoft.com/net/download).

## Install the Quantum Development Kit for Your Preferred Environment ##

### Visual Studio Editors and the Command Line ###

> With the .NET Core SDK now installed, you can implement Q# programs in editors such as Visual Studio Code and Visual Studio, or even directly in the command line.

Reviewer: ```suggestion
After installing the .NET Core SDK, you can create Q# programs in Visual Studio Code and Visual Studio, or using the command line.
```

> With the .NET Core SDK now installed, you can implement Q# programs in editors such as Visual Studio Code and Visual Studio, or even directly in the command line.

Reviewer: I would remove this line, since .NET core does not allow Q# development, they need to install the QDK first

Reviewer 2: For users writing host programs in C#, installing the .NET Core SDK is enough in some sense, though I would definitely recommend installing any applicable IDE extensions and the project templates. The latter should be made a bit less necessary by the New Project... command that was added in Microsoft/qsharp-compiler#79, but it's still helpful.

> To get started, install the Quantum Development Kit following these instructions for your preferred environment (links to download the corresponding editors are included within):

Reviewer: ```suggestion
After installing the [.NET Core SDK](https://dotnet.microsoft.com/), install the Quantum Development Kit following these instructions for your preferred environment:
```

* [Installing the QDK for Visual Studio Code](xref:microsoft.quantum.install.vs-code) (Windows 10, Linux, macOS)
* [Installing the QDK for Visual Studio](xref:microsoft.quantum.install.vs-2017) (Windows 10 only)
* [Installing the QDK in the command line](xref:microsoft.quantum.install.cmd-line)

> ### Other Environments: C#, Python, and Jupyter Notebooks ###

Reviewer: Hmm I wonder if there's a better way to phrase this, or differentiate between language and environment (since this kind of mentions both). We had this conversation a few weeks ago when I made my changes as well, since Python and C# can also be used in VS Code. This is probably fine for now but I want to spend some more time thinking about how we can better differentiate these paths to make it easy for new users.

> If, however, you prefer to work within a different classical host environment, we provide interoperability allowing you to invoke Q# quantum algorithms using C#, Python, and Jupyter Notebooks.

Reviewer: ```suggestion
In addition to the environments mentioned above, we provide interoperability allowing you to invoke Q# quantum algorithms using C#, Python, and Jupyter Notebooks.
```
